src/Reflection/ReflectionType.php

```diff
@@ -43,6 +47,15 @@ public static function createFromNode(
             return new ReflectionIntersectionType($reflector, $owner, $type);
         }
 
+        $nonNullTypes = array_values(array_filter(
```

Comment:
This code is both inefficient and janky as heck: after some thinking, and after experimenting against https://github.com/Roave/BackwardCompatibilityCheck/pull/324, the whole `allowsNull()` is completely elided away, when `ReflectionUnionType::createFromNode()` produces `ReflectionUnionType` for nullable signatures :muscle: